debezium-core/src/main/java/io/debezium/relational/history/KafkaDatabaseHistory.java

```diff
@@ -310,4 +320,31 @@ public String toString() {
     protected static String consumerConfigPropertyName(String kafkaConsumerPropertyName) {
         return CONSUMER_PREFIX + kafkaConsumerPropertyName;
     }
+
+    @Override
+    public void initializeStorage() {
+        super.initializeStorage();
+        final AdminClient admin = AdminClient.create(this.producerConfig.asProperties());
+        try {
+            // Find default replication factor
+            final Collection<Node> nodes = admin.describeCluster().nodes().get(KAFKA_QUERY_TIMEOUT.toMillis(), TimeUnit.MILLISECONDS);
+            if (nodes.isEmpty()) {
+                throw new ConnectException("No brokers available to obtain default settings");
+            }
+            final Map<ConfigResource, Config> configs = admin.describeConfigs(Collections.singleton(new ConfigResource(ConfigResource.Type.BROKER, nodes.iterator().next().idString()))).all().get(KAFKA_QUERY_TIMEOUT.toMillis(), TimeUnit.MILLISECONDS);
+            if (configs.isEmpty()) {
+                throw new ConnectException("No configs have been received");
+            }
+            final Config config = configs.values().iterator().next();
+            final short replicationFactor = Short.parseShort(config.get("default.replication.factor").value());
+
+            // Create topic
+            final NewTopic topic = new NewTopic(topicName, (short)1, replicationFactor);
+            topic.configs(Collect.hashMapOf("cleanup.policy", "delete", "retention.ms", Long.toString(Long.MAX_VALUE)));
+            admin.createTopics(Collections.singleton(topic));
+            logger.info("Database history topic '{}' created", topic);
+        } catch (Exception e) {
+            logger.warn("Creation of database history topic failed, please create the topic manually", e);
```

Comment:
I think this should throw hard exception rather than just a warning. 
Otherwise, the topic will be auto-created anyway.

Reply:
Yeah, makes sense.

Reply:
I falled back to auto-create intentionally so in case something is wrong then first time users do not need to solve it. But the fact is that the net result can be again wrong topic settings in production so it is better to fail fast.